network/src/network.rs

```diff
@@ -220,20 +220,21 @@ impl NetworkState {
             .map(PublicKey::peer_id)
             .expect("Secio must enabled");
 
-        // NOTE: be careful, here easy cause a deadlock,
-        //    because peer_store's lock scope across peer_registry's lock scope
-        let mut peer_store = self.peer_store.lock();
-        let accept_peer_result = {
-            self.peer_registry.write().accept_peer(
+        let accept_peer_result = self.with_peer_registry_mut(|peer_registry| {
+            peer_registry.accept_peer(
                 peer_id.clone(),
                 session_context.address.clone(),
                 session_context.id,
                 session_context.ty,
-                peer_store.as_mut(),
+                self.peer_store.lock().as_mut(),
             )
-        };
+        });
+
         if accept_peer_result.is_ok() {
-            peer_store.update_status(&peer_id, Status::Connected);
+            self.peer_store
+                .lock()
+                .as_mut()
+                .update_status(&peer_id, Status::Connected);
```

Comment:
I'm not sure, do we need consistency here? 
@jjyr

Reply:
`update_status` is a relatively very fast operation, IMO the code may be slower because the 3 times lock acquisition, I do not see any benefit we can get here.

Reply:
Oh I see, `accept_peer` is also very fast, I' think we do not need to worry here.

Reply:
Can we close this PR if not strong reasons to merge? @u2

Reply:
Just be more readable than before. If there are other reasons for the objection, it's no problem to close it.

Reply:
The one strong reason to do not to change here is @zhangsoledad mentioned, in the current implementation the inconsistent maybe not cause big problems, but the change put a potential bug in the code, future refactoring may change the inner behavior and reactivate the bug unintentionally.

And IMO the old code is clear enough.

Reply:
ok, I will close it.